torch_xla/csrc/init_python_bindings.cpp

```diff
@@ -205,18 +205,14 @@ std::shared_ptr<torch::lazy::Value> AllReduceInPlace(
                                  scale, replica_groups, pin_layout));
 }
 
-std::pair<at::Tensor, std::shared_ptr<torch::lazy::Value>> AllReduce(
-    const std::string& reduce_type, const at::Tensor& input,
-    const std::shared_ptr<torch::lazy::Value>& token, double scale,
-    const std::vector<std::vector<int64_t>>& replica_groups, bool pin_layout) {
-  XLATensorPtr result;
-  torch::lazy::Value new_token;
-  std::tie(result, new_token) = tensor_methods::all_reduce(
-      bridge::GetXlaTensor(input), *token, GetReduceType(reduce_type), scale,
-      replica_groups, pin_layout);
-  return std::pair<at::Tensor, std::shared_ptr<torch::lazy::Value>>(
-      bridge::AtenFromXlaTensor(std::move(result)),
-      std::make_shared<torch::lazy::Value>(new_token));
+at::Tensor AllReduce(const std::string& reduce_type, const at::Tensor& input,
+                     double scale,
+                     const std::vector<std::vector<int64_t>>& replica_groups,
+                     bool pin_layout) {
```

Comment:
One thing I am worried about here is that now all_reduce is using a different token than other cc ops.

This means all of the `all_reduce` has a forced dependency and all other cc ops has a force dependency. All_reduce and other cc ops's order will not be enforced. I am OK with this being a short term thing but we do need to fix this before the release.

Reply:
Why? They are the same token from the C++ cache.

Reply:
oh actually nvm, all ops uses c++ token now, my bad.